### DISCUSSION TITLE
How do you handle collision performance?

### DISCUSSION THREAD

**Discussion by lenny991** [Trouv/bevy_ecs_ldtk#336] [Q&A]
Hey, I have been looking around but have come empty handed with how to handle tile collision performantly. I'm thinking of doing something like Unity's CompositeCollider2D but have absolutely no clue on how to do it, and tilemaps with collisions are such a commonly used feature I'd be surprised if nobody has done it. 

I'm using Avian2D for collision (because the creator of it is from the same country than me :D) but this question extends to any physics plugin, I'm willing to switch to Rapier and any other one if they have better support for this kind of system. 

Any assets, plugins, or copypasteable code I can do or a flip I can switch and I'll be happy :D. I tried to implement a system like this but it really just made my head hurt, which made me think if using Rust and Bevy really even is a good idea if I just want to make a video game instead of worry about implementation of different basic game features. Anyway. Thanks in advance!
Lenni

**Comment by Trouv** [comment]
Sorry for the very late response.

The platformer example actually does do collision for the tilemap with some optimization. I.e., it's not one collision per wall tile, but instead combines rectangles of wall tiles into bigger rectangle colliders in a close-to-optimal way. But it uses `rapier`, not avian.

Here's the system that does it:
https://github.com/Trouv/bevy_ecs_ldtk/blob/main/examples/platformer/walls.rs#L16-L183

I'm not exactly sure about the details of unit's `CompositeCollider2D` or if it's significantly more optimal than this, I think most users making platformers w/ this plugin are doing something to the effect of that system though.

**Reply by Trouv** [reply]
Created an issue for documenting this better: #382
